libsql-server/src/database/schema.rs

```diff
@@ -52,12 +52,15 @@ impl crate::connection::Connection for SchemaConnection {
         } else {
             check_program_auth(&ctx, &migration, &self.config.get()).await?;
             let connection = self.connection.clone();
-            validate_migration(&mut migration)?;
+            let disable_foreign_key = validate_migration(&mut migration)?;
             let migration = Arc::new(migration);
             let builder = tokio::task::spawn_blocking({
                 let migration = migration.clone();
                 move || {
                     let res = connection.with_raw(|conn| -> crate::Result<_> {
+                        if disable_foreign_key {
+                            conn.execute("PRAGMA foreign_keys=off", ())?;
+                        }
                         let mut txn = conn
                             .transaction_with_behavior(rusqlite::TransactionBehavior::Immediate)
                             .map_err(|_| {
```

Comment:
If this fails with error we won't re-enable foreign_keys, will we?

Reply:
this is per-connection, the connection will be dropped on error, and foreign key check will be re-enable fo the next connection